### ISSUE TITLE
TASK: Fix PHP 8 Deprecation warnings

### ISSUE THREAD

**Issue by mficzel** [neos/neos-development-collection#3619]
On recent php versions Neos shows some deprecation warnings that should be addressed.

```
PHP Deprecated: Optional parameter $junkCallback declared before required parameter $options is implicitly treated as a required parameter in /var/www/html/Packages/Neos/Neos.Diff/Classes/SequenceMatcher.php on line 83
PHP Deprecated: Optional parameter $arguments declared before required parameter $renderingContext is implicitly treated as a required parameter in /var/www/html/Packages/Neos/Neos.Neos/Classes/ViewHelpers/Backend/IfModuleAccessibleViewHelper.php on line 66
PHP Deprecated: Optional parameter $tidy declared before required parameter $packageKey is implicitly treated as a required parameter in /var/www/html/Packages/Neos/Neos.Neos/Classes/Domain/Service/SiteExportService.php on line 91
PHP Deprecated: Optional parameter $tidy declared before required parameter $pathAndFilename is implicitly treated as a required parameter in /var/www/html/Packages/Neos/Neos.Neos/Classes/Domain/Service/SiteExportService.php on line 119
```

### Affected Versions

PHP: 8+ 
Neos: All versions

**Comment by lorenzulrich** [comment]
Couldn't this also go into all supported versions (Neos 5.3+)?

**Comment by lorenzulrich** [comment]
Well, technically it could be hard to say it's not breaking in any situation, because e.g. someone could have extended the `IfModuleAccessibleViewHelper`. Not sure how to handle that.

**Comment by kitsunet** [comment]
yep, seems one of those edge cases but I would say master / 8.0 it is, and people that are annoyed by the warnings can update ;)

**Comment by lorenzulrich** [comment]
"Can update" will not be a solution in many cases (if you're only working with LTS), but I guess it will become yet another composer patch ;-)

**Comment by mficzel** [comment]
The Changes are merged to 7.0 ... still need upmerges and releases
